pageserver/src/timelines.rs

```diff
@@ -273,31 +273,65 @@ fn run_initdb(conf: &'static PageServerConf, initdbpath: &Path) -> Result<()> {
     Ok(())
 }
 
-//
-// - run initdb to init temporary instance and get bootstrap data
-// - after initialization complete, remove the temp dir.
-//
-fn bootstrap_timeline<R: Repository>(
-    conf: &'static PageServerConf,
-    tenantid: ZTenantId,
-    tli: ZTimelineId,
-    repo: &R,
-) -> Result<()> {
-    let _enter = info_span!("bootstrapping", timeline = %tli, tenant = %tenantid).entered();
-
-    let initdb_path = conf.tenant_path(&tenantid).join("tmp");
+/// Find for uninitialized timelines and try to fix them.
+pub fn find_and_fix_unintialized_timelines(
+    repo: &LayeredRepository,
+    registration_queue: &[ZTimelineId],
+) -> anyhow::Result<()> {
+    let tenant_id = repo.tenant_id();
+
+    for timeline_id in registration_queue {
+        let inmem_timeline = repo.get_timeline_load(*timeline_id).with_context(|| {
```

Comment:
`get_disk_consistent_lsn` is retrieved from the metadata contents during the timeline load.

How realistic is the scenario, that we're able to _load_ the timeline into memory that has some garbage in its metadata? 
Does not look like a feasible check at least, we can read this file directly 🙂

Reply:
> How realistic is the scenario, that we're able to load the timeline into memory that has some garbage in its metadata?

I'm not sure about this actually. As I mentioned in the PR description,

> Right now, there is no reliable way to create an uninitialized timeline without changing the underlying codes. This PR is implemented based on the assumption as described in the original issue (https://github.com/neondatabase/neon/issues/1663).

When replicating #1663, you will get into the scenario that the new timeline folder only contains a `metadata` file, whose data looks like this:
```
TimelineMetadata { 
  hdr: ..., 
  body: TimelineMetadataBody { 
    disk_consistent_lsn: 0/0, 
    prev_record_lsn: None, 
    ancestor_timeline: None, 
    ancestor_lsn: 0/0, 
    latest_gc_cutoff_lsn: 0/16B59D8, 
    initdb_lsn: 0/16B59D8 
   } 
}
```

The "zero" `disk_consistent_lsn` happens when the initialization process failed to checkpoint the import data.